File: pathplannerlib/src/test/java/com/pathplanner/lib/path/PathConstraintsTest.java
package com.pathplanner.lib.path;

import static org.junit.jupiter.api.Assertions.assertEquals;

import edu.wpi.first.math.util.Units;
import org.json.simple.JSONObject;
import org.junit.jupiter.api.Test;

public class PathConstraintsTest {
  public static final double DELTA = 1e-3;

  @Test
  public void testGetters() {
    PathConstraints constraints = new PathConstraints(1.0, 2.0, 3.0, 4.0);

    assertEquals(1.0, constraints.maxVelocityMPS(), DELTA);
    assertEquals(2.0, constraints.maxAccelerationMPSSq(), DELTA);
    assertEquals(3.0, constraints.maxAngularVelocityRadPerSec(), DELTA);
    assertEquals(4.0, constraints.maxAngularAccelerationRadPerSecSq(), DELTA);
  }

  @Test
  public void testFromJson() {
    JSONObject json = new JSONObject();
    json.put("maxVelocity", 1.0);
    json.put("maxAcceleration", 2.0);
    json.put("maxAngularVelocity", 90.0);
    json.put("maxAngularAcceleration", 180.0);
    json.put("nominalVoltage", 12.0);
    json.put("unlimited", false);

    PathConstraints fromJson = PathConstraints.fromJson(json);

    assertEquals(
        new PathConstraints(
            1.0, 2.0, Units.degreesToRadians(90), Units.degreesToRadians(180), 12.0, false),
        fromJson);
  }
}

File: pathplannerlib/src/test/native/cpp/pathplanner/lib/path/PathConstraintsTest.cpp
#include <gtest/gtest.h>
#include "pathplanner/lib/path/PathConstraints.h"

using namespace pathplanner;

TEST(PathConstraintsTest, TestGetters) {
	PathConstraints constraints(1_mps, 2_mps_sq, 3_rad_per_s, 4_rad_per_s_sq);

	EXPECT_DOUBLE_EQ(1.0, constraints.getMaxVelocity()());
	EXPECT_DOUBLE_EQ(2.0, constraints.getMaxAcceleration()());
	EXPECT_DOUBLE_EQ(3.0, constraints.getMaxAngularVelocity()());
	EXPECT_DOUBLE_EQ(4.0, constraints.getMaxAngularAcceleration()());
}

TEST(PathConstraintsTest, TestFromJson) {
	wpi::json json;
	json.emplace("maxVelocity", 1.0);
	json.emplace("maxAcceleration", 2.0);
	json.emplace("maxAngularVelocity", 90.0);
	json.emplace("maxAngularAcceleration", 180.0);
	json.emplace("nominalVoltage", 12.0);
	json.emplace("unlimited", false);

	PathConstraints fromJson = PathConstraints::fromJson(json);
	EXPECT_EQ(PathConstraints(1_mps, 2_mps_sq, 90_deg_per_s, 180_deg_per_s_sq, 12_V, false), fromJson);
}